### PR TITLE
Bypass sign-in in offline mode

### PR DESCRIPTION
**Pull request by ethangraham2001**
# Changes

- When app boots offline, it will sign in using `DEFAULT_OFFLINE_PROFILE` value. Liking itineraries has no effect in offline mode. It is important that the app boots in offline mode - if the app launches and then is put into airplane mode, sign-in will still try to connect to firebase auth. I don't have a good solution for this as of right now.
- `LoginViewModel` now takes network status (boolean) to decide whether or not to attempt connection through firebase auth
- `ProfileRepositoryImpl` now takes context; this allows us to check network status on the fly and decide whether or not to try and fetch data from firestore
- Updated tests for compatibility with new context parameter.

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/data/ProfileRepositoryImpl.kt [hunk 1]
```diff
@@ -123,6 +125,16 @@ class ProfileRepositoryImpl(db: FirebaseFirestore) : ProfileRepository {
   }
 
   override suspend fun checkIfItineraryIsLiked(userUid: String, itineraryUid: String): Boolean {
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/data/ProfileRepositoryImpl.kt:127]: lgtm

File: app/src/main/java/com/github/wanderwise_inc/app/ui/signin/LoginScreen.kt [hunk 2]
```diff
@@ -23,19 +23,28 @@ import androidx.compose.ui.unit.dp
...
+    navController: NavController
+) {
   val signInState by loginViewModel.signInState.observeAsState()
 
   LaunchedEffect(signInState) {
     if (signInState == SignInState.SUCCESS) {
       navController.navigate(Graph.HOME)
+    } else if (signInState == SignInState.OFFLINE) {
+      profileViewModel.setActiveProfile(DEFAULT_OFFLINE_PROFILE)
+      navController.navigate(Graph.HOME)
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/signin/LoginScreen.kt:47]: lgtm

File: app/src/test/java/com/github/wanderwise_inc/app/data/ProfileRepositoryTest.kt [hunk 3]
```diff
@@ -50,10 +60,23 @@ class ProfileRepositoryTest {
...
+    networkInfo = mock()
+    networkCapabilities = mock()
+
     `when`(db.collection(anyString())).thenReturn(userCollection)
-    profileRepositoryImpl = ProfileRepositoryImpl(db)
+    `when`(context.getSystemService(anyString())).thenReturn(connectivityManager)
+    `when`(connectivityManager.activeNetwork).thenReturn(networkInfo)
+    `when`(connectivityManager.getNetworkCapabilities(networkInfo)).thenReturn(networkCapabilities)
+    `when`(networkCapabilities.hasCapability(anyInt())).thenReturn(true)
+  }
```
Comment (ThomasPFMuller) [inline app/src/test/java/com/github/wanderwise_inc/app/data/ProfileRepositoryTest.kt:74]: lgtm, seems like mocking the network connectivity query was a pain. Good work

**Review by ThomasPFMuller — APPROVED** [review]
Great work, well done

File: app/src/main/java/com/github/wanderwise_inc/app/data/ProfileRepositoryImpl.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/data/ProfileRepositoryImpl.kt:1]: Good way to check if the network is available before doing some tasks.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/graph/AuthNavGraph.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/graph/AuthNavGraph.kt:1]: Lgtm

File: app/src/main/java/com/github/wanderwise_inc/app/ui/signin/LoginScreen.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/signin/LoginScreen.kt:1]: Clean way to handle offline connection, this was a major task

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/LoginViewModel.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/LoginViewModel.kt:1]: Lgtm well done

File: app/src/test/java/com/github/wanderwise_inc/app/data/ProfileRepositoryTest.kt (hunk 3 above)
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/data/ProfileRepositoryTest.kt:74]: Lgtm, those look like easy mocks 😎

**Pull request merged**